platforms/core-execution/build-cache-http/src/main/java/org/gradle/caching/http/HttpBuildCache.java

```diff
@@ -55,49 +58,22 @@
  */
 public abstract class HttpBuildCache extends AbstractBuildCache {
     private final HttpBuildCacheCredentials credentials;
-    private URI url;
-    private boolean allowUntrustedServer;
-    private boolean allowInsecureProtocol;
-    private boolean useExpectContinue;
 
     public HttpBuildCache() {
         this.credentials = getObjectFactory().newInstance(HttpBuildCacheCredentials.class);
+        getAllowUntrustedServer().convention(false);
+        getAllowInsecureProtocol().convention(false);
+        getUseExpectContinue().convention(false);
     }
 
     @Inject
-    protected ObjectFactory getObjectFactory() {
-        throw new UnsupportedOperationException();
-    }
+    protected abstract ObjectFactory getObjectFactory();
 
     /**
      * Returns the URI to the cache.
      */
-    @Nullable
-    @Nested
-    public URI getUrl() {
-        return url;
-    }
-
-    /**
-     * Sets the URL of the cache. The URL must end in a '/'.
-     */
-    public void setUrl(String url) {
-        setUrl(URI.create(url));
-    }
-
-    /**
-     * Sets the URL of the cache. The URL must end in a '/'.
-     */
-    public void setUrl(URL url) throws URISyntaxException {
-        setUrl(url.toURI());
-    }
-
-    /**
-     * Sets the URL of the cache. The URL must end in a '/'.
-     */
-    public void setUrl(@Nullable URI url) {
-        this.url = url;
-    }
+    @ReplacesEagerProperty(adapter = UrlAdapter.class)
+    public abstract Property<URI> getUrl();
 
     /**
      * Returns the credentials used to access the HTTP cache backend.
```

Comment:
❗ It's not strictly related to the migration, but we should replace the whole logic around `credentials`, too. There should be no need to inject `ObjectFactory`, and indeed, we should only have:

```java
@Nested
public abstract HttpBuildCacheCredentials getCredentials();
```

...and the `credentials(Action)` method.